services/motion/builtin/builtin.go

```diff
@@ -308,23 +315,30 @@ func (ms *builtIn) MoveOnGlobe(
 
 		// once execution responds: return the result to the caller
 		case resp := <-ma.responseChan:
-			ms.logger.Debugf("execution completed: %s", resp)
+			ms.logger.Debugf("execution response: %s", resp)
 			ma.cancel()
+			if errors.Is(resp.err, context.Canceled) {
```

Comment:
Why is this getting added?

Reply:
While unlikely, it is possible for us to issue a replan once the plan has been fully executed and we are checking if we are close enough to the goal with the `deviatedFromPlan` method.

In this case, we have `err != nil` so we return `moveResponse{err: err}` from `moveRequest.execute`. This then bubbles up into the `select` cases so it must be handled accordingly.

Similarly, the `responseChan` for the obstacle and position polling routines may return an error which is `context.Cancelled`.

Reply:
How would we get a context cancelled error here that we wouldn't want to return, though? A call to `ma.cancel()` should flush all channels, so if we get a context cancelled here, it would be a real cancellation by the user. So it would loop again and just hit the return on line 301.

If `deviatedFromPlan` deems a replan is in order after plan execution is complete, it'll return `false, nil`, not a context cancelled error; so this would want to check `resp.success`.

Reply:
I am imagining the following scenario.

Suppose we have only one waypoint left and the robot has some non-nil errorState but is within planDeviationMeters.

Suppose that as we call `GoToInputs` on the final set of inputs we also call `obstaclesIntersectPlan`.
Suppose we find a collision right when `GoToInputs` successfully returns.
My reasoning is that by the time we call `deviatedFromPlan` from within the `execute` method, the error of `deviatedFromPlan` should be `context.Cancelled`.

Perhaps I am missing something since there are a lot of moving parts.
Lmk your thoughts.

Reply:
A call to `ma.cancel()` includes flushing the channels:
```
	utils.FlushChan(ma.position.responseChan)
	utils.FlushChan(ma.obstacle.responseChan)
	utils.FlushChan(ma.responseChan)
```
so that context cancelled should be dealt with there and shouldn't surface here. Is this an issue you observed happening?

Reply:
yes, this was something I was observing but I am now unable to recreate it. I will remove `if errors.Is(resp.err, context.Canceled) {`.